rlclientlib/live_model_impl.cc

```diff
@@ -69,13 +71,15 @@ namespace reinforcement_learning {
       RETURN_IF_FAIL(explore_exploit(event_id, context, response, status));
     }
     response.set_event_id(event_id);
-    RETURN_IF_FAIL(_ranking_logger->log(event_id, context, flags, response, status));
+    RETURN_IF_FAIL(_ranking_logger->log(event_id, context, flags, response, status, _decision_mode));
 
     // Check watchdog for any background errors. Do this at the end of function so that the work is still done.
     if (_watchdog.has_background_error_been_reported()) {
       RETURN_ERROR_LS(_trace_logger.get(), status, unhandled_background_error_occurred);
     }
 
+    RETURN_IF_FAIL(post_process_rank(response, _decision_mode));
```

Comment:
>post_process_rank [](start = 19, length = 17)

Trainer need to train using following tuple for unbiased training
[a,p(a),c(a),x]
a = user chosen
p(a) = 1
c(a)=feedback from env
x = features

In this case why is local inference even needed?

Reply:
As discussed offline, VW could not learn from 0 probability. This is by design.